src/fidesctl/ctl/connectors/aws.py

```diff
@@ -162,3 +176,80 @@ def create_rds_systems(
         if not instance.get("DBClusterIdentifier")
     ]
     return rds_cluster_systems + rds_instances_systems
+
+
+def create_resource_tagging_systems(
+    resource_arns: List[str],
+    organization_key: str,
+) -> List[System]:
+    """
+    Given a list of resource arns, build a list of systems object which represents
+    each resource.
+    """
+    resource_generators = {
+        "dynamodb": create_tagging_dynamodb_system,
+        "s3": create_tagging_s3_system,
+    }
+    systems = []
+    for arn in resource_arns:
+        arn_split = arn.split(":")
+        arn_resource_type = arn_split[2]
+        resource_generator = resource_generators.get(arn_resource_type)
+        if resource_generator:
+            generated_system = resource_generator(arn, organization_key)
+            if generated_system:
+                systems.append(generated_system)
+    return systems
+
+
+def create_tagging_dynamodb_system(
+    arn: str,
+    organization_key: str,
+) -> Optional[System]:
+    """
+    Given an AWS arn for a dynamodb resource, returns a System representation
+    if System is desired.
+    """
+    arn_split = arn.split(":")
+    resource_name = arn_split[5]
+
+    if resource_name.startswith("table/"):
+        table_name = resource_name[len("table/") :]
+        system = System(
+            fides_key=table_name,
+            name=table_name,
+            description=f"Fides Generated Description for DynamoDb table: {table_name}",
+            system_type="dynamodb_table",
+            organization_fides_key=organization_key,
+            fidesctl_meta=SystemMetadata(
+                resource_id=arn,
+            ),
+            privacy_declarations=[],
+        )
+    return system
+
+
+def create_tagging_s3_system(
+    arn: str,
+    organization_key: str,
+) -> Optional[System]:
+    """
+    Given an AWS arn for a s3 resource, returns a System representation
+    if System is desired.
```

Comment:
what does `if System is desired` mean in this context?

Reply:
An aws service could have different resource types. So for instance for dynamodb we only create systems for tables but not necessarily for auto scaling policies.

Reply:
Updating the comment to make this more clear!

Reply:
ohhh got it! thanks for the clarification